examples/gamma/main.rs

```diff
@@ -50,18 +52,43 @@ pub fn main() {
         .with_title("Gamma example".to_string())
         .with_dimensions(1024, 768)
         .with_vsync();
-    let (window, mut device, mut factory, main_color, mut main_depth) =
-        gfx_window_glutin::init::<ColorFormat, DepthFormat>(builder, &events_loop);
-    let mut encoder: gfx::Encoder<_, _> = factory.create_command_buffer().into();
+    let window = gfx_window_glutin::build(builder, &events_loop, ColorFormat::get_format(), DepthFormat::get_format());
+
+    let (mut surface, adapters) = gfx_window_glutin::Window(&window).get_surface_and_adapters();
+    let queue_descs = adapters[0].get_queue_families().iter()
+                                 .filter(|family| surface.supports_queue(&family) )
+                                 .map(|family| { (family, 1) })
+                                 .collect::<Vec<_>>();
+
+    // Open device (factory and queues)
+    let gfx::Device { mut factory, mut general_queues, mut graphics_queues, .. } = adapters[0].open(&queue_descs);
```

Comment:
I just understand correctly what you meant with the second option (in particular with removing 10 lines). Imo the advantage of this is too small to add it as additional function. In order to remove the queue cast below we would need to change the queue family, which we could consider in the future.

Just requesting a `GraphicsCommandQueue` could be conflicting with some backends like GL. The current approach I'm aiming at for GL is to either export a `GraphicsQueue` or `GeneralQueue` depending on compute support.